sklearn/tests/test_naive_bayes.py

```diff
@@ -536,3 +537,45 @@ def test_naive_bayes_scale_invariance():
               for f in [1E-10, 1, 1E10]]
     assert_array_equal(labels[0], labels[1])
     assert_array_equal(labels[1], labels[2])
+
+
+def test_alpha():
+    # Setting alpha=0 should not output nan results when p(x_i|y_j)=0 is a case
+    X = np.array([[1, 0], [1, 1]])
+    y = np.array([0, 1])
+    nb = BernoulliNB(alpha=0.)
+    assert_warns(UserWarning, nb.partial_fit, X, y, classes=[0, 1])
+    assert_warns(UserWarning, nb.fit, X, y)
+    prob = np.array([[1, 0], [0, 1]])
+    assert_array_almost_equal(nb.predict_proba(X), prob)
+
+    nb = MultinomialNB(alpha=0.)
+    assert_warns(UserWarning, nb.partial_fit, X, y, classes=[0, 1])
+    assert_warns(UserWarning, nb.fit, X, y)
+    prob = np.array([[2./3, 1./3], [0, 1]])
+    assert_array_almost_equal(nb.predict_proba(X), prob)
+
+    # Test sparse X
+    X = scipy.sparse.csr_matrix(X)
+    nb = BernoulliNB(alpha=0.)
+    nb.fit(X, y)
```

Comment:
This will also raise the warning which we would rather not see in tests. Either assert or ignore.